src/diffusers/loaders/ip_adapter.py

```diff
@@ -205,10 +205,12 @@ def unload_ip_adapter(self):
             self.image_encoder = None
             self.register_to_config(image_encoder=[None, None])
 
-        # remove feature extractor
-        if hasattr(self, "feature_extractor") and getattr(self, "feature_extractor", None) is not None:
-            self.feature_extractor = None
-            self.register_to_config(feature_extractor=[None, None])
+        # remove feature extractor only when safety_checker is None as safety_checker uses
+        # the feature_extractor later
+        if hasattr(self, "safety_checker") and getattr(self, "safety_checker", None) is None:
```

Comment:
```suggestion
        if not hasattr(self, "safety_checker"):
```



when we disable `safety_checker` in a pipeline, e.g. when we do 
```python
pipe = DiffusionPipeline.from_pretrained("runwayml/stable-diffusion-v1-5", safety_checker=None)
``` 

we do not remove the `feature_extractor` component. So I think we should not unload `feature_extractor` in `unload_ip_adapter` just because `safety_checker` is `None`. 

We should unload `feature_extrator` only when the `feature_extractor` is loaded by ip_adapter. I think when the pipeline does not have a `safety_checker` component at all (SDXL pipelines), we know it is loaded by ip_adapter, no?

Reply:
Makes sense.